### ISSUE TITLE
libressl fails to build on aarch64-darwin

### ISSUE THREAD

**Issue by LoganBarnett** [NixOS/nixpkgs#181187]
### Steps To Reproduce
1. Build `libressl` via `nix-shell -p libressl` (might need to be from `aarch64-darwin`).
2. Observe test 68 `rc4test` fail.

### Build log
https://gist.github.com/LoganBarnett/0bfb5d35fed0509b7fd6cdf424fe6afc

### Additional context
I can build `libressl_3_4` just fine, though that's just for additional context. I'm not suggesting a rollback.

#180404 might be when the change was introduced. I don't actually know if my system was using 3.5.x prior to this build attempt. 

Thanks for helping maintain this project!

### Notify maintainers
@thoughtpolice 
@fpletz
@ajs124 (not a maintainer, but the contributor for #180404).

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.14 MiB unpacked):
  /nix/store/072148x7wcqa62cdzpgixzlm3ip96df7-DarwinTools-1
  /nix/store/vy75zc398lw109im2xfhzk5qha7xnbad-nix-info
copying path '/nix/store/072148x7wcqa62cdzpgixzlm3ip96df7-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/vy75zc398lw109im2xfhzk5qha7xnbad-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.4.0, macOS 12.3.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.6.1`
 - channels(logan): `"home-manager, nixpkgs"`
 - channels(root): `"home-manager, nixpkgs"`
 - nixpkgs: `/Users/logan/.nix-defexpr/channels/nixpkgs`
```

**Comment by ajs124** [comment]
This seems aarch64-darwin specific, yes. `aarch64-linux` and `x86_64-darwin` seem to work just fine.

**Comment by jiegec** [comment]
Related issue: https://github.com/libressl-portable/portable/issues/749

**Comment by jiegec** [comment]
Fixed by pr #181239
